warehouse/warehouse.go

```diff
@@ -819,43 +1126,93 @@ func (wh *HandleT) processStagingFile(job LoadFileJobT) (loadFileIDs []int64, er
 	return
 }
 
-func getBucketFolder(job LoadFileJobT, tableName string) string {
-	job.TableToBucketFolderMapLock.Lock()
-	defer job.TableToBucketFolderMapLock.Unlock()
+// Gets the config from config backend and extracts enabled writekeys
+func monitorDestRouters() {
+	ch := make(chan utils.DataEvent)
+	backendconfig.Subscribe(ch, "backendConfig")
+	dstToWhRouter := make(map[string]*HandleT)
 
-	folderName, ok := job.TableToBucketFolderMap[tableName]
-	if !ok {
-		folderName = fmt.Sprintf(`%v-%v`, strconv.FormatInt(job.Upload.ID, 10), uuid.NewV4().String())
-		job.TableToBucketFolderMap[tableName] = folderName
-		return folderName
-	}
-	return folderName
-}
+	for {
+		config := <-ch
+		logger.Debug("Got config from config-backend", config)
+		sources := config.Data.(backendconfig.SourcesT)
+		enabledDestinations := make(map[string]bool)
+		for _, source := range sources.Sources {
+			if source.Enabled {
+				for _, destination := range source.Destinations {
+					if destination.Enabled {
```

Comment:
We need to enable for all destinations even if they are not enabled.

Reply:
We should do the same for source too (process irrespective of they are enabled)? @lokey

Reply:
removed both. will also check with @lokey